pkg/controllers/allocation/filter.go

```diff
@@ -63,6 +58,36 @@ func (f *Filter) GetProvisionablePods(ctx context.Context, provisioner *v1alpha2
 	return provisionable, nil
 }
 
+// GetProvisionerFor retrieves the provisioner responsible for the pod
+func (f *Filter) GetProvisionerFor(ctx context.Context, p *v1.Pod) (*v1alpha2.Provisioner, error) {
```

Comment:
I might move this up a level to controller, since that's the only place its used and the cohesion with the rest of the filtering logic feels off.